### PR TITLE
CHE-995: discover ssh machines architecture automatically

### PR DESCRIPTION
**Pull request by garagatyi**
Discover ssh machines architecture automatically.
Remove beta field in machine conf that provides architecture.
Move server-side classes to appropriate package.
Removed workarounds in ssh client and properly wait until command finishes.

### CODE REVIEW

**Comment by garagatyi** [comment]
@skabashnyuk @vparfonov Please review

File: plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/SshMachineImplTerminalLauncher.java [hunk 1]
```diff
@@ -61,39 +72,106 @@ public String getMachineType() {
...
+                                                               null);
+        ListLineConsumer lineConsumer = new ListLineConsumer();
+        getUnameOutput.start(lineConsumer);
+        String unameOutput = lineConsumer.getText();
+        Matcher matcher = UNAME_OUTPUT.matcher(unameOutput);
+        if (matcher.matches()) {
+            String os = matcher.group("os").toLowerCase();
+            String arch = matcher.group("architecture").toLowerCase();
+            StringBuilder result = new StringBuilder();
+            if (os.contains("linux")) {
```
Comment (skabashnyuk) [inline plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/SshMachineImplTerminalLauncher.java:131]: Shall we add this string constants to some enum?

File: plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/jsch/JschSshClient.java [hunk 2]
```diff
@@ -236,64 +237,47 @@ private String getAbsolutePath(String path) throws MachineException {
...
-            // todo fix that
-            // this method is used to run commands such as "pwd" or "mkdir -p /some/path"
-            // And we need to wait to check if it return 0. If command is still running method returns -1.
-            try {
-                // workaround exit code -1
-                Thread.sleep(1000);
-            } catch (InterruptedException ignored) {
-            }
-            return exec.getExitStatus();
+            IoUtil.readStream(inStream);
```
Comment (skabashnyuk) [inline plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/jsch/JschSshClient.java:250]: why we need to read this streams?

**Comment by skabashnyuk** [comment]
other ok

File: plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/SshMachineImplTerminalLauncher.java (hunk 1 above)
Comment (garagatyi) [inline plugins/plugin-ssh-machine/src/main/java/org/eclipse/che/plugin/machine/ssh/SshMachineImplTerminalLauncher.java:131]: I don't see the reason to add constant with value equal to name

**Pull request merged**